pkg/provider/acme/local_store.go

```diff
@@ -81,7 +87,10 @@ func (s *LocalStore) get(resolverName string) (*StoredData, error) {
 				}
 				if len(certificates) < len(storedData.Certificates) {
 					storedData.Certificates = certificates
-					s.saveDataChan <- s.storedData
+
+					// we cannot pass s.storedData directly, map is reference type and as result
+					// we can face with race condition, so we need to work with objects copy
+					s.saveDataChan <- s.unSafeCopyOfStoredData(s.storedData)
 				}
```

Comment:
To follow up my first comment.

```suggestion

					// we cannot pass s.storedData directly, map is reference type and as result
					// we can face with race condition, so we need to work with objects copy
					s.saveDataChan <- s.copyOfStoredData()
				}
```

Reply:
@rtribotte In this case, I think, we will have a deadlock because at line 41/47 we already made locking.